packages/react/src/hooks/useClaimsTransaction.ts

```diff
@@ -0,0 +1,93 @@
+import { Address, Hex } from "viem";
+import { useClaims } from "./useClaims";
+import { useSimulateContract } from "wagmi";
+import { ADDRESS, SupportedChain } from "@metrom-xyz/contracts";
+import { metromAbi } from "@metrom-xyz/contracts/abi";
+import { useMemo } from "react";
+import {
+    SimulateContractErrorType,
+    SimulateContractReturnType,
+} from "wagmi/actions";
+import { Claim } from "@metrom-xyz/sdk";
+
+interface UseClaimsTransactionParams {
+    chainId: SupportedChain;
+    address?: Address;
+}
+
+interface UseClaimsTransactionReturnValue {
+    loading: boolean;
+    error: SimulateContractErrorType | null;
+    claims: Claim[];
+    transaction?: SimulateContractReturnType<
+        typeof metromAbi,
+        "claimRewards",
+        [
+            {
+                campaignId: Hex;
+                proof: Hex[];
+                token: Address;
+                amount: bigint;
+                receiver: Address;
+            }[],
+        ]
+    >;
+}
+
+/**
+ * Fetches the available claims for an account and simulates/validates the contract interaction
+ * in order to perform the rewards claim.
+ *
+ * @param {Object} param - The parameters object.
+ * @param {string} param.address - The wallet address of the receiver account.
+ * @param {SupportedChain} param.chainId - The chain id.
+ *
+ * @returns {UseClaimsTransactionReturnValue} Object including the simulation result, that can be used to submit the transaction,
+ * using the useWriteContract hook from wagmi.
+ */
+export function useClaimsTransaction({
+    address,
+    chainId,
```

Comment:
I think it might be better for the caller to pass in the claims here. It's a bit confusing because the `useClaims` hooks fetches claims across all chains, while this one is only applied to a single chains. I think the caller might have more freedom if they instead fetched the claims, filtered them based on the rule they wanted (e.g. by chain) and then passed them here. We just need to check for the claims to be all on the same chain here and we should be ok. Opinions?

Reply:
Alternatively, it might actually be better for the hook to receive ALL the claims and a chain id. Then we'd just remove the `useClaims` call here and we should be good.

Reply:
Makes sense, addressed in a823d357b01221c0c0abb3ed5651b3ed47663090.